server/src/main/java/org/candlepin/audit/ListenerWrapper.java

```diff
@@ -14,60 +14,75 @@
+ * Event handling. It is responsible for translating the message to an Event object,
+ * and passing it along to a candlepin EventListener for processing.
+ *
+ * The ListenerWrapper handles all client session management and ensures that the
+ * message remains on the message queue until it has been successfully processed by
+ * the listener.
  */
 public class ListenerWrapper implements MessageHandler {
 
     private EventListener listener;
     private static Logger log = LoggerFactory.getLogger(ListenerWrapper.class);
     private ObjectMapper mapper;
-    public ListenerWrapper(EventListener listener, ObjectMapper mapper) {
+    private ClientSession session;
+
+    public ListenerWrapper(EventListener listener, ObjectMapper mapper, ClientSession session) {
         this.listener = listener;
         this.mapper = mapper;
+        this.session = session;
     }
 
     @Override
     public void onMessage(ClientMessage msg) {
-        String body = msg.getBodyBuffer().readString();
-        log.debug("Got event: {}", body);
-
-        // Exceptions thrown here will cause the event to remain in ActiveMQ:
+        String body = "";
         try {
+            // Acknowledge the message so that the server knows that it was received.
+            // By doing this, the server can update the delivery counts which plays
+            // part in calculating redelivery delays.
+            msg.acknowledge();
+            log.debug("ActiveMQ message {} acknowledged for listener: {}", msg.getMessageID(), listener);
+
+            // Process the message via our EventListener framework.
+            body = msg.getBodyBuffer().readString();
+            log.debug("Got event: {}", body);
             Event event = mapper.readValue(body, Event.class);
             listener.onEvent(event);
-        }
-        catch (JsonMappingException e) {
-            log.error("Unable to deserialize event object from msg: " + body, e);
-            throw new RuntimeException("Error deserializing event", e);
-        }
-        catch (JsonParseException e) {
-            log.error("Unable to deserialize event object from msg: " + body, e);
-            throw new RuntimeException("Error deserializing event", e);
-        }
-        catch (IOException e) {
-            log.error("Unable to deserialize event object from msg: " + body, e);
-            throw new RuntimeException("Error deserializing event", e);
-        }
 
-        try {
-            msg.acknowledge();
-            log.debug("ActiveMQ message acknowledged for listener: " + listener);
+            // Finally commit the session so that the message is taken out of the queue.
+            session.commit();
         }
-        catch (ActiveMQException e) {
-            log.error("Unable to acknowledge ActiveMQ msg", e);
+        catch (Exception e) {
+            // Log a warning instead of a full stack trace to reduce log size.
+            String messageId = msg == null ? "" : Long.toString(msg.getMessageID());
```

Comment:
Can we put these ternary predicates in parenthesis?

```
String messageId = (msg == null) ? "" : Long.toString(msg.getMessageID());
```

Reply:
:+1: